diff --git a/core/modules/system/src/EventSubscriber/ConfigCacheTag.php b/core/modules/system/src/EventSubscriber/ConfigCacheTag.php
index 6a4d807..6a2740f 100644
--- a/core/modules/system/src/EventSubscriber/ConfigCacheTag.php
+++ b/core/modules/system/src/EventSubscriber/ConfigCacheTag.php
@@ -47,22 +47,25 @@ public function __construct(ThemeHandlerInterface $theme_handler, CacheTagsInval
    *   The Event to process.
    */
   public function onSave(ConfigCrudEvent $event) {
+    $saved_config = $event->getConfig();
+
     // Changing the site settings may mean a different route is selected for the
     // front page. Additionally a change to the site name or similar must
     // invalidate the render cache since this could be used anywhere.
-    if ($event->getConfig()->getName() === 'system.site') {
+    if ($saved_config->getName() === 'system.site') {
       $this->cacheTagsInvalidator->invalidateTags(['route_match', 'rendered']);
     }
 
     // Theme configuration and global theme settings.
-    if (in_array($event->getConfig()->getName(), ['system.theme', 'system.theme.global'], TRUE)) {
+    if (\in_array($saved_config->getName(), ['system.theme', 'system.theme.global'], TRUE) &&
+      strpos($saved_config->get('default'), 'os_ct_') === FALSE) {
       $this->cacheTagsInvalidator->invalidateTags(['rendered']);
     }
 
     // Theme-specific settings, check if this matches a theme settings
     // configuration object, in that case, clear the rendered cache tag.
     foreach (array_keys($this->themeHandler->listInfo()) as $theme_name) {
-      if ($theme_name == $event->getConfig()->getName()) {
+      if ($theme_name == $saved_config->getName()) {
         $this->cacheTagsInvalidator->invalidateTags(['rendered']);
         break;
       }
diff --git a/core/modules/system/src/SystemConfigSubscriber.php b/core/modules/system/src/SystemConfigSubscriber.php
index 0ab0d15..47f88eb 100644
--- a/core/modules/system/src/SystemConfigSubscriber.php
+++ b/core/modules/system/src/SystemConfigSubscriber.php
@@ -39,7 +39,9 @@ public function __construct(RouteBuilderInterface $router_builder) {
    */
   public function onConfigSave(ConfigCrudEvent $event) {
     $saved_config = $event->getConfig();
-    if ($saved_config->getName() == 'system.theme' && ($event->isChanged('admin') || $event->isChanged('default'))) {
+
+    // Rebuild routes only if the default theme is not a custom theme.
+    if (($saved_config->getName() === 'system.theme' && strpos($saved_config->get('default'), 'os_ct_') === FALSE) && ($event->isChanged('admin') || $event->isChanged('default'))) {
       $this->routerBuilder->setRebuildNeeded();
     }
   }
