diff --git a/modules/bibcite_entity/bibcite_entity.links.contextual.yml b/modules/bibcite_entity/bibcite_entity.links.contextual.yml
new file mode 100644
index 0000000..ec83e79
--- /dev/null
+++ b/modules/bibcite_entity/bibcite_entity.links.contextual.yml
@@ -0,0 +1,9 @@
+entity.bibcite_reference.edit_form:
+  route_name: entity.bibcite_reference.edit_form
+  group: bibcite_reference
+  title: Edit
+
+entity.bibcite_reference.delete_form:
+  route_name: entity.bibcite_reference.delete_form
+  group: bibcite_reference
+  title: Delete
diff --git a/modules/bibcite_entity/src/ReferenceViewBuilder.php b/modules/bibcite_entity/src/ReferenceViewBuilder.php
index 4918f0d..41c02aa 100644
--- a/modules/bibcite_entity/src/ReferenceViewBuilder.php
+++ b/modules/bibcite_entity/src/ReferenceViewBuilder.php
@@ -70,6 +70,11 @@ class ReferenceViewBuilder extends EntityViewBuilder {
           '#theme' => 'bibcite_citation',
           '#data' => $this->serializer->normalize($entity, 'csl'),
           '#entity' => $entity,
+          '#contextual_links' => [
+            'bibcite_reference' => [
+              'route_parameters' => ['bibcite_reference' => $entity->id()],
+            ],
+          ],
         ];
       }
     }
diff --git a/modules/bibcite_entity/templates/bibcite-reference.html.twig b/modules/bibcite_entity/templates/bibcite-reference.html.twig
index 71a408d..f9cd329 100644
--- a/modules/bibcite_entity/templates/bibcite-reference.html.twig
+++ b/modules/bibcite_entity/templates/bibcite-reference.html.twig
@@ -4,6 +4,8 @@
  * Default theme implementation to present Reference data.
  *
  * Available variables:
+ * - title_suffix: Additional output populated by modules, intended to be
+ *   displayed after the main title tag that appears in the template.
  * - content: All node items. Use {{ content }} to print them all,
  *   or print a subset such as {{ content.field_example }}. Use
  *   {{ content|without('field_example') }} to temporarily suppress the printing
@@ -18,6 +20,7 @@
  */
 #}
 <div{{ attributes }}>
+  {{ title_suffix }}
   {% if content %}
     {{ content }}
   {% endif %}
