diff --git a/plugins/context_reaction_menu.inc b/plugins/context_reaction_menu.inc
index 0d82adc..80e5f98 100644
--- a/plugins/context_reaction_menu.inc
+++ b/plugins/context_reaction_menu.inc
@@ -138,6 +138,6 @@ class context_reaction_menu extends context_reaction {
       list($menu_name, $path) = explode(':', $id);
       $by_menu_name[$menu_name][] = $path;
     }
-    return $by_menu_name;
+    return $active_paths;
   }
 }
