diff --git a/src/Element/Registrants.php b/src/Element/Registrants.php
index b087c5a..6dbc0c1 100644
--- a/src/Element/Registrants.php
+++ b/src/Element/Registrants.php
@@ -186,7 +186,7 @@ class Registrants extends FormElement {
     $element['for_arity'] = [
       '#type' => 'radios',
       '#title' => t('This registration is for'),
-      '#options' => NULL,
+      '#options' => [],
       '#access' => $for_arity_any_arity && $change_it,
       '#attributes' => [
         'class' => ['for_arity'],
diff --git a/src/Form/EventTypeForm.php b/src/Form/EventTypeForm.php
index c538c45..30e4925 100644
--- a/src/Form/EventTypeForm.php
+++ b/src/Form/EventTypeForm.php
@@ -120,7 +120,7 @@ class EventTypeForm extends EntityForm {
         $form['#attached']['library'][] = 'rng/rng.admin';
         $form['entity_type'] = [
           '#type' => 'radios',
-          '#options' => NULL,
+          '#options' => [],
           '#title' => $this->t('Event entity type'),
           '#required' => TRUE,
         ];
@@ -139,7 +139,7 @@ class EventTypeForm extends EntityForm {
           '#description' => $this->t('Use an existing entity/bundle combination.'),
           '#return_value' => "existing",
           '#parents' => array('entity_type'),
-          '#default_value' => '',
+          '#default_value' => [],
         ];
 
         $form['entity_type']['existing']['container'] = [
diff --git a/src/Form/RegistrationForm.php b/src/Form/RegistrationForm.php
index 65d9c77..8a485c5 100644
--- a/src/Form/RegistrationForm.php
+++ b/src/Form/RegistrationForm.php
@@ -131,7 +131,7 @@ class RegistrationForm extends ContentEntityForm {
         '#type' => 'checkbox',
         '#title' => $this->t('Create new revision'),
         '#description' => $this->t('Revisions record changes between saves.'),
-        '#default_value' => FALSE,
+        '#default_value' => [],
         '#access' => $current_user->hasPermission('administer rng'),
         '#group' => 'revision_information',
       ];
diff --git a/src/Plugin/Menu/LocalAction/ResetAccessRules.php b/src/Plugin/Menu/LocalAction/ResetAccessRules.php
index 4a67235..28c6bc4 100644
--- a/src/Plugin/Menu/LocalAction/ResetAccessRules.php
+++ b/src/Plugin/Menu/LocalAction/ResetAccessRules.php
@@ -8,6 +8,7 @@ use Drupal\rng\EventManagerInterface;
 use Drupal\Core\Routing\RouteMatchInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\HttpFoundation\Request;
 
 /**
  * Modifies the reset access rules action.
@@ -70,7 +71,7 @@ class ResetAccessRules extends LocalActionDefault {
   /**
    * {@inheritdoc}
    */
-  public function getTitle() {
+  public function getTitle(Request $request = NULL) {
     $route = $this->routeProvider->getRouteByName($this->getRouteName());
     $param = $route->getDefault('event');
 
